core/tests/reflection/type.cpp

```diff
@@ -34,5 +34,12 @@ TEST_CASE("reflection::Type")
         CHECK(type.get<int>() == 42);
     }
 
+    SUBCASE("with two traits")
+    {
+        auto& anotherType = type;
+        REQUIRE(anotherType == type);
+        Type::destroy(anotherType);
```

Comment:
```suggestion
        auto& anotherType = type;
        REQUIRE(anotherType == type);
```
Destroy is already being called on line 44. This is a double delete error.
Also, would be cool to also test inequality, e.g., create another type with a different name and assert that == returns false.